.github/workflows/build-and-push-sync-server.yml

```diff
@@ -0,0 +1,55 @@
+name: build-and-push-sync-server
+on:
+  push:
+    branches:
+    - master
+    - develop
```

Comment:
IMO develop branches shouldn't be deployed to DockerHub because they might be unstable. I normally recommend using a separate Docker registry like Github Packages. See my comments here: https://github.com/ankicommunity/anki-devops-services/issues/11#issuecomment-787505446.